src/lib/flash-helper.js

```diff
@@ -174,11 +175,31 @@ async function createFlashSteps({ modules, isInDfuMode, platformId }) {
 			flashStep.flashMode = 'normal';
 			normalModules.push(flashStep);
 		} else {
+			if (moduleType === 'userPart') {
+				deviceOsVersion = module.prefixInfo.depModuleVersion;
+			}
 			flashStep.flashMode = 'dfu';
 			dfuModules.push(flashStep);
 		}
 	});
 
+	const DEVICE_OS_MIN_VERSION_TO_FORMAT_128K_USER = 3103;
+	// invalidate 128k user part for device-os 3.1 and later
+	if (platform.dfu.segments.formerUserPart && deviceOsVersion >= DEVICE_OS_MIN_VERSION_TO_FORMAT_128K_USER) {
+		const moduleInfo = {
+			prefixInfo : {
+				moduleStartAddy : platform.dfu.segments.formerUserPart.address
+			}
+		};
+		const formerUserPartflashStep = {
+			name: 'invalidate-128k-user-part',
+			moduleInfo,
+			data: Buffer.alloc(platform.dfu.segments.formerUserPart.size, 0xFF)
+		};
+		formerUserPartflashStep.flashMode = 'dfu';
```

Comment:
You can include `flashMode: 'dfu'` in the object initialization instead of a separate line in this case